contracts/libraries/ArbitraryMessage.sol

```diff
@@ -0,0 +1,82 @@
+pragma solidity 0.4.24;
+
+import "../interfaces/IBridgeValidators.sol";
+import "./Message.sol";
+
+library ArbitraryMessage {
+    // layout of message :: bytes:
+    // offset  0: 32 bytes :: uint256 - message length
+    // offset 32: 20 bytes :: address - sender address
+    // offset 52: 20 bytes :: address - executor contract
+    // offset 72: 32 bytes :: bytes32 txHash
```

Comment:
The reason to inject the transactions hash into the message is to have an unique message that is mark as processed so it can't be processed again. The same message could be generated from an user multiple times so adding the transactions hash make it possible to continue processing the new messages.  Also this parameter is included on the events `AffirmationCompleted` and `RelayedMessage` for traceability.

I first considered to add the `txHash` at the end of the message, but to avoid calculations in `unpackData` method to know in which position the `txHash` will be, since it will depend on the `dataType`, I added it in the middle similar to the message that is constructed based on `UserRequestForSignature` as you mention. Adding the `txHash` at the beginning is a good option too, If you think it will be better I can change it

Reply:
> The reason to inject the transactions hash into the message is to have an unique message that is mark as processed so it can't be processed again.

Yes, I know this.

> Adding the txHash at the beginning is a good option too, If you think it will be better I can change it

Let's do this in order to simplify the oracles' work to compose the message.

Reply:
Update message to include `txHash` in the beginning https://github.com/poanetwork/poa-bridge-contracts/pull/77/commits/1987b9ab285577ef263bece6488b48e204879455